src/devices/video/mc6845.cpp

```diff
@@ -450,15 +470,15 @@ void mc6845_device::recompute_parameters(bool postload)
 		vsync_off_pos = vert_pix_total;
 
 	/* update only if screen parameters changed, unless we are coming here after loading the saved state */
-	if (postload ||
+	if ((postload ||
 		(horiz_pix_total != m_horiz_pix_total) || (vert_pix_total != m_vert_pix_total) ||
-		(max_visible_x != m_max_visible_x) || (max_visible_y != m_max_visible_y) ||
+		(visible_width != m_visible_width) || (visible_height != m_visible_height) ||
 		(hsync_on_pos != m_hsync_on_pos) || (vsync_on_pos != m_vsync_on_pos) ||
-		(hsync_off_pos != m_hsync_off_pos) || (vsync_off_pos != m_vsync_off_pos))
+		(hsync_off_pos != m_hsync_off_pos) || (vsync_off_pos != m_vsync_off_pos)) && !maintain_visible_area)
```

Comment:
You still want to reconfigure the screen if you’re maintaining the visible area, because the totals or sync positions could have changed.